### PR TITLE
Add full coverage for LiquidationStrategy contract

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

**Review by Anyhowclick — DISMISSED** [review]

**Review by Anyhowclick — DISMISSED** [review]

**Review by ducquangkstn — COMMENTED** [review]
Not in this PR but
- IPool contract is lack of documentation
- INoSwappingLiquidationStrategy.sol seems redundant to me

File: contracts/treasury/LiquidationStrategy.sol [hunk 1]
```diff
@@ -68,6 +63,8 @@ contract LiquidationStrategy is ILiquidationStrategy, PermissionAdmin, Utils, Re
     _setWhitelistedLiquidatorsEnabled(false);
   }
 
+  receive() external payable {}
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply contracts/treasury/LiquidationStrategy.sol:66]: seems to be added when doing rebase

File: contracts/treasury/LiquidationStrategy.sol [hunk 2]
```diff
@@ -288,7 +287,7 @@ contract LiquidationStrategy is ILiquidationStrategy, PermissionAdmin, Utils, Re
     uint64 _repeatedPeriod,
     uint64 _duration
   ) internal {
-    // TODO: Validate
+    require(_repeatedPeriod > 0, 'repeatedPeriod == 0');
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply contracts/treasury/LiquidationStrategy.sol:290]: we can set _duration >= _repatedPeriod to enable "always allow" liquidation process

**Review by Anyhowclick — APPROVED** [review]

File: contracts/mock/MockLiquidatorWithCallback.sol [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.6;
+
+import {IERC20Ext} from '@kyber.network/utils-sc/contracts/IERC20Ext.sol';
+import {Utils} from '@kyber.network/utils-sc/contracts/Utils.sol';
+import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import {MockLiquidationStrategy} from './MockLiquidationStrategy.sol';
+
+contract MockLiquidatorWithCallback is Utils {
```
Comment (ducquangkstn) [inline contracts/mock/MockLiquidatorWithCallback.sol:9]: Pls implement `contracts/interfaces/liquidation/ILiquidationCallback.sol`

File: contracts/mock/MockLiquidatorWithCallback.sol [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  function setTransferBackAmount(uint256 amount) external {
+    transferBackAmount = amount;
+  }
+
+  function setTestReentrancy(bool shouldTest) external {
+    shouldTestReentrancy = shouldTest;
+  }
+
+  function liquidationCallback(
```
Comment (ducquangkstn) [inline contracts/mock/MockLiquidatorWithCallback.sol:27]: Thinking about this again and it would be better if we can pass minReturn amount as params

**Review by ducquangkstn — APPROVED** [review]

**Pull request merged**